Gems/Atom/RPI/Code/Source/RPI.Public/RenderPipeline.cpp

```diff
@@ -470,8 +470,6 @@ namespace AZ
                 }
                 else if (pipelineViews.m_type == PipelineViewType::Persistent)
                 {
-                    // Reset persistent view: clean draw list mask and draw lists
-                    pipelineViews.m_views[0]->Reset();
```

Comment:
why remove this? it's just clearing the draw list tag and the various draw list data pieces, I don't see any deallocation happening.

Reply:
View reset will reset the draw list context which will clear the array of arrays which means the capacity for all the arrays gets zeroed and adding new draw items will result in the usual re-allocations as the arrays grow.

Reply:
Since clear doesn't reset the vector's m_end parameter the capacity is unchanged, m_last is changed so all the entries are removed.

Reply:
The issue is that DrawListsByTag is an array of vectors. So if you clear the array, the vectors get destroyed, not cleared.

Reply:
Array doesn't have a clear function, it's actually calling clear on all the contained vectors.

Reply:
My bad. It's actually a vector of unique_ptrs to arrays of vectors. So when the vector of unique_ptrs is cleared, the arrays are released.